src/Sylius/Bundle/ImportExportBundle/Form/Type/ExportProfileType.php

```diff
@@ -0,0 +1,143 @@
+ */
+class ExportProfileType extends AbstractResourceType
+{
+    /**
+     * Reader registry.
+     *
+     * @var ServiceRegistryInterface
+     */
+    protected $readerRegistry;
+
+    /**
+     * Writer registry.
+     *
+     * @var ServiceRegistryInterface
+     */
+    protected $writerRegistry;
+
+    /**
+     * Constructor.
+     *
+     * @param ServiceRegistryInterface $readerRegistry
+     * @param ServiceRegistryInterface $writerRegistry
+     */
+    public function __construct($dataClass, array $validationGroups, ServiceRegistryInterface $readerRegistry, ServiceRegistryInterface $writerRegistry)
+    {
+        parent::__construct($dataClass, $validationGroups);
+
+        $this->readerRegistry = $readerRegistry;
+        $this->writerRegistry = $writerRegistry;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->addEventSubscriber(new BuildReaderFormListener($this->readerRegistry, $builder->getFormFactory()))
+            ->addEventSubscriber(new BuildWriterFormListener($this->writerRegistry, $builder->getFormFactory()))
+            ->add('name', 'text', array(
+                'label' => 'sylius.form.export_profile.name',
+                'required' => true,
+            ))
+            ->add('code', 'text', array(
+                'label'    => 'sylius.form.export_profile.code',
+                'required' => true,
+            ))
+            ->add('description', 'textarea', array(
+                'label'    => 'sylius.form.export_profile.description',
+                'required' => false,
+            ))
+            ->add('reader', 'sylius_export_reader_choice', array(
+                'label'    => 'sylius.form.reader.name',
+            ))
+            ->add('writer', 'sylius_export_writer_choice', array(
+                'label' => 'sylius.form.writer.name',
+            ))
+        ;
+
+        $prototypes = array(
+            'readers' => array(),
+            'writers' => array(),
+        );
+
+        foreach ($this->readerRegistry->all() as $type => $reader) {
+            $formType = sprintf('sylius_%s_reader', $reader->getType());
+
+            if (!$formType) {
+                continue;
+            }
+
+            try {
+                $prototypes['readers'][$type] = $builder->create('readerConfiguration', $formType)->getForm();
+            } catch (\InvalidArgumentException $e) {
+                continue;
+            }
+        }
+
+        foreach ($this->writerRegistry->all() as $type => $writer) {
+            $formType = sprintf('sylius_%s_writer', $writer->getType());
+
+            if (!$formType) {
+                continue;
+            }
+
+            try {
+                $prototypes['writers'][$type] = $builder->create('writerConfiguration', $formType)->getForm();
+            } catch (\InvalidArgumentException $e) {
+                continue;
+            }
+        }
```

Comment:
I would try to make some private function for common for those two foreach loops, but it's only me ;)